homeassistant/components/homee/entity.py

```diff
@@ -0,0 +1,126 @@
+"""Base Entities for Homee integration."""
+
+from pyHomee.const import AttributeType, NodeProfile
+from pyHomee.model import HomeeAttribute, HomeeNode
+
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import Entity
+
+from . import HomeeConfigEntry
+from .const import DOMAIN
+from .helpers import get_name_for_enum
+
+
+class HomeeNodeEntity(Entity):
+    """Representation of a Node in Homee."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, node: HomeeNode, entry: HomeeConfigEntry) -> None:
+        """Initialize the wrapper using a HomeeNode and target entity."""
+        self._node = node
+        self._clear_node_listener = None
+        self._attr_unique_id = node.id
+        self._entry = entry
+
+        self._homee_data = {
+            "id": node.id,
+            "name": node.name,
+            "profile": node.profile,
+            "attributes": [{"id": a.id, "type": a.type} for a in node.attributes],
+        }
+        self._attr_device_info = dr.DeviceInfo(
+            identifiers={(DOMAIN, self._node.id)},
+            name=self._node.name,
+            manufacturer="unknown",
+            model=get_name_for_enum(NodeProfile, self._homee_data["profile"]),
+            sw_version=self._get_software_version(),
+            via_device=(DOMAIN, self._entry.runtime_data.settings.uid),
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Add the homee binary sensor device to home assistant."""
+        self.async_on_remove(self._node.add_on_changed_listener(self._on_node_updated))
+
+    @property
+    def available(self) -> bool:
+        """Return the availability of the underlying node."""
+        return self._node.state <= 1
+
+    @property
+    def should_poll(self) -> bool:
+        """Return if the entity should poll."""
+        return False
+
+    @property
+    def raw_data(self):
+        """Return the raw data of the node."""
+        return self._node.raw_data
+
+    async def async_update(self):
+        """Fetch new state data for this node."""
+        # Base class requests the whole node, if only a single attribute is needed
+        # the platform will overwrite this method.
+        homee = self._entry.runtime_data
+        await homee.update_node(self._node.id)
+
+    def attribute(self, attribute_type):
+        """Try to get the current value of the attribute of the given type."""
+        try:
+            attribute = self._node.get_attribute_by_type(attribute_type)
+        except KeyError:
+            raise AttributeNotFoundException(attribute_type) from None
+
+        # If the unit of the attribute is 'text', it is stored in .data
+        if attribute.unit == "text":
+            return self._node.get_attribute_by_type(attribute_type).data
+
+        return self._node.get_attribute_by_type(attribute_type).current_value
```

Comment:
Would it make sense to have this as logic in the library?

Reply:
Yes